src/v2/users/pagination.rs

```diff
@@ -2,15 +2,28 @@ use std::{future::Future, pin::Pin, task::Poll};
 
 use anyhow::{Context, Result};
 use futures::{FutureExt, Stream};
-use serde::{de::DeserializeOwned, Serialize};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
 use url::Url;
 
-use crate::v2::Zitadel;
+use crate::v2::{users, Zitadel};
 
-pub(crate) trait PaginationResponse {
-	type Item;
+#[derive(Debug, Serialize, Deserialize, Default, Clone)]
+pub struct PaginatedResponse<T> {
+	#[serde(rename = "details")]
+	details: Option<users::ListDetails>,
+	#[serde(rename = "sortingColumn")]
+	sorting_column: Option<users::UserFieldName>,
```

Comment:
The `UserFieldName` is specific to the list users response, other responses might return other columns
```rust
pub struct PaginatedResponse<T, S=String> {
	#[serde(rename = "details")]
	details: Option<users::ListDetails>,
	#[serde(rename = "sortingColumn")]
	sorting_column: Option<S>,
	#[serde(rename = "result")]
	result: Option<Vec<T>>,
}
```
and then instantiate it to `PaginatedResponse<User, UserFieldName>` somewhere in the code:
```rust
pub fn list_users(&mut self, body: ListUsersRequest) -> Result<(impl Stream<Item = User>, Option<UserFieldName>)> {..}
async fn get_more_data<T: DeserializeOwned + 'static, S: Deserialize>(
	mut zitadel: Zitadel,
	query: impl Serialize + Send,
	endpoint: Url,
) -> Result<(Vec<T>, S)> {..}
```
 Or since we don't actually use it and return `impl Stream<Item=T>` we can just remove it completely

Reply:
You are right, we don't use it and the deserialization doesn't fail if the json has more fields. Now it should be better. But I still have the Send problem, even making `pub struct PaginatedResponse<T: Send>`

Reply:
Constraining type parameters in type definition (even if possible) is anti pattern (`DatatypeContexts` from haskell was deprecated for this reason)